Update dyninst to v13

pranswarup opened the pull request:
Update Dyninst submodule
Refactoring of build scripts to build TBB, Boost, ElfUtils, and LibIberty, since Dyninst build scripts no longer do.
Workflows are now building Dyninst and its dependencies.

pranswarup committed:
Added Boost, TBB, ElfUtils & LibIberty to rocprofiler-systems build system, Work in progress for parallel build

pranswarup committed:
Parallel build complete, WIP install

pranswarup committed:
Added files to find external dependencies

pranswarup committed:
Added FindLibDwarf and modified elfutils find

pranswarup committed:
Installation of externals in the appropriate folder complete

pranswarup committed:
Remove extra boost find package added by mistake

pranswarup committed:
remove dyninst cmake functions from upper level dyninst boost cmake

pranswarup committed:
Remove else conditions to make it compatible with ci builds

pranswarup committed:
Add external build options to rocprofiler-systems build

pranswarup committed:
Add external build options to rocprofiler-systems github workflows builds

pranswarup committed:
Add external build options to rocprofiler-systems github workflows build jobs

pranswarup committed:
Revert to original TBB version for compatibility with workflow builds

pranswarup committed:
format cmake, change some workflow options to build dyninst where possible

pranswarup committed:
format packages.cmake

pranswarup committed:
To link to external dyninst add the missing target Dyninst::Boost_headers

pranswarup committed:
Remove extra backslashes from opensuse.yml

pranswarup committed:
Upgrade TBB to compatible version

pranswarup committed:
Updated the docker ci script to pull from dyninst_13_external branch to prebuild dyninst

pranswarup committed:
Cmake is not handling automatic conversion from ElfUtils_DOWNLOAD_VERSION to ELFUTILS_DOWNLOAD_VERSION in CMAKE so handle it manually

pranswarup committed:
remove FindElfUtils.cmake

pranswarup committed:
Revert TBB source to orignal

pranswarup committed:
Replace DDYNINST with DROCPROFSYS in build-release.sh script

pranswarup committed:
Update git submodule to latest commit

pranswarup committed:
Merge remote-tracking branch 'origin/amd-staging' into rocprof-tbb-boost-etc

pranswarup committed:
Added a workaround for a bug in RCCL resulting in double free or corruption error when exiting

marantic-amd commented on CMakeLists.txt:
```diff
@@ -9,6 +9,11 @@ if(CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR AND CMAKE_CURRENT_SOURCE_DIR STREQ
 
+# find_package() uses upper-case <PACKAGENAME>_ROOT variables.
+if(POLICY CMP0144)
+    cmake_policy(SET CMP0144 NEW)
```
```suggestion
    cmake_policy(SET ${POLICY} NEW)
```

marantic-amd commented on cmake/Packages.cmake:
```diff
@@ -329,17 +330,37 @@ if(ROCPROFSYS_BUILD_DYNINST)
-        COMPONENT dyninst
-        FILES_MATCHING
-        PATTERN "*${CMAKE_SHARED_LIBRARY_SUFFIX}*")
+    foreach(
```
It's better practice to declare `_LIB` variable beforehand. Just a clearer cmake file and better readability.

marantic-amd commented:
Just some minor style corrections, apart from that looks LGMT.

pranswarup committed:
Build dyninst with all github workflows

pranswarup commented:
Turned on building dyninst for all workflow as there are path issues when building docker ci containers with pre-built dyninst.

dgaliffiAMD commented on .github/workflows/redhat.yml:
```diff
@@ -43,7 +43,7 @@ jobs:
     runs-on: ubuntu-latest
     container:
-      image: dgaliffiamd/rocprofiler-systems:ci-base-rhel-${{ matrix.os-release }}
+      image: swaruppranjal/test:ci-base-rhel-${{ matrix.os-release }}
```
Don't forget to remove your test repo.

pranswarup replied:
This is a temporary commit. I will revert it upon testing.

pranswarup replied:
Reverted.

dgaliffiAMD commented on .github/workflows/opensuse.yml:
```diff
@@ -43,7 +43,7 @@ jobs:
     runs-on: ubuntu-latest
     container:
-      image: dgaliffiamd/rocprofiler-systems:ci-base-opensuse-${{ matrix.os-release }}
+      image: swaruppranjal/test:ci-base-opensuse-${{ matrix.os-release }}
```
Don't forget to remove your test repo.

dgaliffiAMD commented on .github/workflows/ubuntu-focal.yml:
```diff
@@ -43,7 +43,7 @@ jobs:
     runs-on: ubuntu-latest
     container:
-      image: dgaliffiamd/rocprofiler-systems:ci-base-ubuntu-20.04
+      image: swaruppranjal/test:ci-base-ubuntu-20.04
```
Don't forget to remove your test repo.

dgaliffiAMD commented on .github/workflows/ubuntu-focal.yml:
```diff
@@ -244,7 +249,7 @@ jobs:
     runs-on: ubuntu-latest
     container:
-      image: dgaliffiamd/rocprofiler-systems:ci-base-ubuntu-20.04
+      image: swaruppranjal/test:ci-base-ubuntu-20.04
```
Don't forget to remove your test repo.

dgaliffiAMD commented on .github/workflows/ubuntu-focal.yml:
```diff
@@ -406,7 +416,7 @@ jobs:
 
     container:
-      image: dgaliffiamd/rocprofiler-systems:ci-base-ubuntu-20.04
+      image: swaruppranjal/test:ci-base-ubuntu-20.04
```
Don't forget to remove your test repo.

dgaliffiAMD commented on .github/workflows/ubuntu-jammy.yml:
```diff
@@ -43,7 +43,7 @@ jobs:
     runs-on: ubuntu-latest
     container:
-      image: dgaliffiamd/rocprofiler-systems:ci-base-ubuntu-22.04
+      image: swaruppranjal/test:ci-base-ubuntu-22.04
```
Don't forget to remove your test repo.

dgaliffiAMD commented on .github/workflows/ubuntu-jammy.yml:
```diff
@@ -133,6 +133,10 @@ jobs:
+          -DROCPROFSYS_BUILD_TBB=ON
+          -DROCPROFSYS_BUILD_ELFUTILS=ON
+          -DROCPROFSYS_BUILD_LIBIBERTY=ON
           -DROCPROFSYS_BUILD_HIDDEN_VISIBILITY=${{ matrix.hidden }}
```
```suggestion
          -DROCPROFSYS_BUILD_BOOST=${{ matrix.build-dyninst }}
          -DROCPROFSYS_BUILD_TBB=${{ matrix.build-dyninst }}
          -DROCPROFSYS_BUILD_ELFUTILS=${{ matrix.build-dyninst }}
          -DROCPROFSYS_BUILD_LIBIBERTY=${{ matrix.build-dyninst }}
          -DROCPROFSYS_BUILD_HIDDEN_VISIBILITY=${{ matrix.hidden }}
```

dgaliffiAMD commented on .github/workflows/ubuntu-noble.yml:
```diff
@@ -43,7 +43,7 @@ jobs:
     runs-on: ubuntu-latest
     container:
-      image: dgaliffiamd/rocprofiler-systems:ci-base-ubuntu-24.04
+      image: swaruppranjal/test:ci-base-ubuntu-24.04
```
Don't forget to remove your test repo.

dgaliffiAMD commented on .github/workflows/ubuntu-jammy.yml:
```diff
@@ -310,6 +314,10 @@ jobs:
+          -DROCPROFSYS_BUILD_BOOST=ON
+          -DROCPROFSYS_BUILD_TBB=ON
+          -DROCPROFSYS_BUILD_ELFUTILS=ON
+          -DROCPROFSYS_BUILD_LIBIBERTY=ON
```
```suggestion
          -DROCPROFSYS_BUILD_BOOST=${{ matrix.build-dyninst }}
          -DROCPROFSYS_BUILD_TBB=${{ matrix.build-dyninst }}
          -DROCPROFSYS_BUILD_ELFUTILS=${{ matrix.build-dyninst }}
          -DROCPROFSYS_BUILD_LIBIBERTY=${{ matrix.build-dyninst }}
```

dgaliffiAMD commented on cmake/DyninstBoost.cmake:
```diff
@@ -0,0 +1,380 @@
+                                   ON)
+    # If we didn't find a suitable version on the system, then download one from the web
+    rocprofiler_systems_add_cache_option(
+        BOOST_DOWNLOAD_VERSION "Version of boost to download and install" STRING "1.79.0")
```
```suggestion
        ROCPROFSYS_BOOST_DOWNLOAD_VERSION "Version of boost to download and install" STRING "1.79.0")
```

dgaliffiAMD commented on cmake/DyninstBoost.cmake:
```diff
@@ -0,0 +1,380 @@
+    rocprofiler_systems_add_cache_option(
+        BOOST_DOWNLOAD_VERSION "Version of boost to download and install" STRING "1.79.0")
+
+    # If the user specifies a version other than BOOST_DOWNLOAD_VERSION, use that version.
+    if(${BOOST_DOWNLOAD_VERSION} VERSION_LESS ${Boost_MIN_VERSION})
+        rocprofiler_systems_message(
+            FATAL_ERROR
+            "Boost download version is set to ${BOOST_DOWNLOAD_VERSION} but Boost minimum version is set to ${Boost_MIN_VERSION}"
+            )
+    endif()
```
```suggestion
    # If the user specifies a version other than ROCPROFSYS_BOOST_DOWNLOAD_VERSION, use that version.
    if(${ROCPROFSYS_BOOST_DOWNLOAD_VERSION} VERSION_LESS ${Boost_MIN_VERSION})
        rocprofiler_systems_message(
            FATAL_ERROR
            "Boost download version is set to ${ROCPROFSYS_BOOST_DOWNLOAD_VERSION} but Boost minimum version is set to ${Boost_MIN_VERSION}"
            )
    endif()
```

dgaliffiAMD commented on cmake/DyninstBoost.cmake:
```diff
@@ -0,0 +1,380 @@
+    endif()
+
+    rocprofiler_systems_message(
+        STATUS "Attempting to build ${BOOST_DOWNLOAD_VERSION} as external project")
```
```suggestion
        STATUS "Attempting to build BOOST(${ROCPROFSYS_BOOST_DOWNLOAD_VERSION}) as external project")
```
Updated log to match style of other components

dgaliffiAMD commented on cmake/DyninstBoost.cmake:
```diff
@@ -0,0 +1,380 @@
+        list(APPEND BOOST_ARGS cflags=-fPIC cxxflags=-fPIC)
+    endif()
+
+    string(REPLACE "." "_" _boost_download_filename ${BOOST_DOWNLOAD_VERSION})
```
```suggestion
    string(REPLACE "." "_" _boost_download_filename ${ROCPROFSYS_BOOST_DOWNLOAD_VERSION})
```

dgaliffiAMD commented on cmake/DyninstBoost.cmake:
```diff
@@ -0,0 +1,380 @@
+        Boost-External
+        PREFIX ${PROJECT_BINARY_DIR}/boost
+        GIT_REPOSITORY https://github.com/boostorg/boost.git
+        GIT_TAG boost-${BOOST_DOWNLOAD_VERSION}
```
```suggestion
        GIT_TAG boost-${ROCPROFSYS_BOOST_DOWNLOAD_VERSION}
```

dgaliffiAMD commented on scripts/build-release.sh:
```diff
@@ -222,7 +222,7 @@ ROCPROFSYS_GENERAL_ARGS="-DROCPROFSYS_CPACK_SYSTEM_NAME=${DISTRO} -DROCPROFSYS_R
 ROCPROFSYS_USE_ARGS="-DROCPROFSYS_USE_MPI_HEADERS=ON -DROCPROFSYS_USE_OMPT=ON -DROCPROFSYS_USE_PAPI=ON"
 TIMEMORY_ARGS="-DTIMEMORY_USE_LIBUNWIND=ON -DTIMEMORY_BUILD_LIBUNWIND=ON -DTIMEMORY_BUILD_PORTABLE=ON"
-DYNINST_ARGS="-DROCPROFSYS_BUILD_DYNINST=ON -DDYNINST_USE_OpenMP=ON $(echo -DDYNINST_BUILD_{TBB,BOOST,ELFUTILS,LIBIBERTY}=ON) -DDYNINST_BOOST_DOWNLOAD_VERSION=${BOOST_VERSION}"
+DYNINST_ARGS="-DROCPROFSYS_BUILD_DYNINST=ON -DROCPROFSYS_USE_OpenMP=ON $(echo -DROCPROFSYS_BUILD_{TBB,BOOST,ELFUTILS,LIBIBERTY}=ON) -DROCPROFSYS_BOOST_DOWNLOAD_VERSION=${BOOST_VERSION}"
```
```suggestion
DYNINST_ARGS="-DROCPROFSYS_BUILD_DYNINST=ON -DROCPROFSYS_USE_OpenMP=ON $(echo -DROCPROFSYS_BUILD_{TBB,BOOST,ELFUTILS,LIBIBERTY}=ON) -DROCPROFSYS_BOOST_DOWNLOAD_VERSION=${BOOST_VERSION}"
```
The CMake module is just using `BOOST_DOWNLOAD_VERSION`. See previous comments/suggestions which adds the "ROCPROFSYS" prefix.

dgaliffiAMD replied:
It doesn't look like DYNINST_USE_OpenMP is being passed down either. Might as well just remove it from this script.

pranswarup replied:
I changed to -DDYNINST_USE_OpenMP=ON as there is a USE_OpenMP variable in dyninst.
https://github.com/ROCm/dyninst/blob/dyninst_13/cmake/tpls/DyninstOpenMP.cmake

dgaliffiAMD replied:
I think we can just remove. It was never used before and I don't want it to be confused for an option that removed "OpenMP" support for rocprof-sys as a whole.

pranswarup replied:
Ok I removed it.

dgaliffiAMD commented on docker/build-docker-ci.sh:
```diff
@@ -134,7 +134,7 @@ if [ ! -f ${DOCKER_FILE} ]; then
 
 verbose-run rm -rf ./dyninst-source
-verbose-run cp -r ../external/dyninst ./dyninst-source
+verbose-run git clone --branch dyninst_13_external https://github.com/ROCm/dyninst.git dyninst-source
```
Still required?

dgaliffiAMD commented on source/lib/rocprof-sys/library.cpp:
I wouldn't include this in _this_ PR.

dgaliffiAMD reviewed (changes requested):
Thanks, Pranjal. I'm still working through it, but I've made a few initial comments. 

Also, there a still a few references to DYNINST_BUILD* throughout the CMake files and documentation. It would be worth doing a string search through the project.

pranswarup committed:
Update CMakeLists.txt

Co-authored-by: marantic-amd <marantic@amd.com>

pranswarup committed:
Update cmake/DyninstBoost.cmake

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

pranswarup committed:
Update cmake/DyninstBoost.cmake

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

pranswarup committed:
Update .github/workflows/ubuntu-jammy.yml

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

pranswarup committed:
Update .github/workflows/ubuntu-jammy.yml

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

pranswarup committed:
Update cmake/DyninstBoost.cmake

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

pranswarup committed:
Update cmake/DyninstBoost.cmake

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

pranswarup committed:
Update cmake/DyninstBoost.cmake

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

pranswarup committed:
Remove reference to external dyninst v13

pranswarup committed:
Revert "Update CMakeLists.txt"

This reverts commit c6eff192bfa2cf9c84884e430392d4acb8b3ce23.

dgaliffiAMD committed:
Format source

Signed-off-by: David Galiffi <David.Galiffi@amd.com>

pranswarup committed:
Revert to -DDYNINST_USE_OpenMP

pranswarup committed:
Remove -DDYNINST_USE_OpenMP

dgaliffiAMD commented on cmake/DyninstExternals.cmake:
```diff
@@ -0,0 +1,112 @@
+include(MacroUtilities)
+
+# Set BUILD_BOOST to ON if ROCPROFSYS_BUILD_BOOST is ON
+if(ROCPROFSYS_BUILD_BOOST)
+    rocprofiler_systems_add_option(BUILD_BOOST "Enable building Boost internally" ON)
+endif()
+
+# Set BUILD_TBB to ON if ROCPROFSYS_BUILD_TBB is ON
+if(ROCPROFSYS_BUILD_TBB)
+    rocprofiler_systems_add_option(BUILD_TBB "Enable building TBB internally" ON)
+endif()
+
+# Set BUILD_ELFUTILS to ON if ROCPROFSYS_BUILD_ELFUTILS is ON
+if(ROCPROFSYS_BUILD_ELFUTILS)
+    rocprofiler_systems_add_option(BUILD_ELFUTILS "Enable building elfutils internally"
+                                   ON)
+endif()
+
+# Set BUILD_LIBIBERTY to ON if ROCPROFSYS_BUILD_LIBIBERTY is ON
+if(ROCPROFSYS_BUILD_LIBIBERTY)
+    rocprofiler_systems_add_option(BUILD_LIBIBERTY "Enable building libiberty internally"
+                                   ON)
+endif()
```
```suggestion
# Map deprecated DYNINST_BUILD_* variables to new ROCPROFSYS_BUILD_* variables
foreach(dep BOOST TBB ELFUTILS LIBIBERTY)
    if(DYNINST_BUILD_${dep})
        message(WARNING "DYNINST_BUILD_${dep} is deprecated. Use ROCPROFSYS_BUILD_${dep} instead.")
        set(ROCPROFSYS_BUILD_${dep} ON)
    endif()
endforeach()

# Set BUILD_* to ON if ROCPROFSYS_BUILD_* is ON
foreach(dep BOOST TBB ELFUTILS LIBIBERTY)
    if(ROCPROFSYS_BUILD_${dep})
        if(dep STREQUAL "BOOST")
            rocprofiler_systems_add_option(BUILD_BOOST "Enable building Boost internally" ON)
        elseif(dep STREQUAL "TBB")
            rocprofiler_systems_add_option(BUILD_TBB "Enable building TBB internally" ON)
        elseif(dep STREQUAL "ELFUTILS")
            rocprofiler_systems_add_option(BUILD_ELFUTILS "Enable building elfutils internally" ON)
        elseif(dep STREQUAL "LIBIBERTY")
            rocprofiler_systems_add_option(BUILD_LIBIBERTY "Enable building libiberty internally" ON)
        endif()
    endif()
endforeach()
```
A small change for backwards compatibility.

pranswarup committed:
Update cmake/DyninstExternals.cmake

Co-authored-by: David Galiffi <David.Galiffi@amd.com>

dgaliffiAMD committed:
Rename Boost target names to match project format

Signed-off-by: David Galiffi <David.Galiffi@amd.com>

dgaliffiAMD committed:
Revert "Added a workaround for a bug in RCCL resulting in double free or corruption error when exiting"

This reverts commit 7e0cbeb330af0c47f9093503894e1951cf6a8ff6.

Signed-off-by: David Galiffi <David.Galiffi@amd.com>

dgaliffiAMD committed:
Rename TBB target names to match project format

Signed-off-by: David Galiffi <David.Galiffi@amd.com>

dgaliffiAMD committed:
Rename LibIberty target names to match project format

Signed-off-by: David Galiffi <David.Galiffi@amd.com>

dgaliffiAMD committed:
Formatting cmake

Signed-off-by: David Galiffi <David.Galiffi@amd.com>

dgaliffiAMD committed:
Move BOOST_ROOT to the external/ folder

dgaliffiAMD committed:
Refactor tbb build folders

dgaliffiAMD committed:
Refactored libiberty's build folders

dgaliffiAMD committed:
Fix TBB lib install

dgaliffiAMD committed:
Refactored elfutil build folders.

Use our existing ElfUtils.cmake module, rather than DyninstElfUtils.cmake

dgaliffiAMD committed:
Update documention and cpack file

Use ROCPROFSYS_BUILD_* rather than DYNINST_BUILD_* for building the Dyninst dependencies

dgaliffiAMD committed:
Format cmake

dgaliffiAMD committed:
Revert "Refactored elfutil build folders."

This reverts commit eb781a07eaf8efdcdd52350201fc0fd11a806a23.
